### ISSUE TITLE
Proposal: A way to make contrived forms

### ISSUE THREAD

**Issue by mholt** [headzoo/surf#9]
Really enjoying surf so far. However, as mentioned in the TODO section of the readme, surf doesn't execute JavaScript. One of my projects right now involves logging in on a site where the login form is injected into the page by JavaScript. I can use the network inspector in Chrome to see what the form submit looks like, but the trouble is getting the form into the DOM for a browser that doesn't support JavaScript.

One thought I had would be to contrive a form that doesn't actually exist on the DOM but is submittable all the same. Basically, the user would invoke `surf.NewFakeForm(action, method)` which acts like a regular surf.Form except that methods like `Input(name, value)` don't check to make sure `name` already exists (it would create it).

Basically, the Form type might become an interface.

Would you accept a pull request that adds this functionality? And I'm still kind of new to this package, so if you know of a better way to accomplish what I need to do, feel free to let me know!
